platforms/native/language-native/src/main/java/org/gradle/language/nativeplatform/internal/incremental/IncrementalCompileProcessor.java

```diff
@@ -38,33 +39,18 @@ public IncrementalCompileProcessor(ObjectHolder<CompilationState> previousCompil
     public IncrementalCompilation processSourceFiles(final Collection<File> sourceFiles) {
         return buildOperationExecutor.call(new CallableBuildOperation<IncrementalCompilation>() {
             @Override
-            public IncrementalCompilation call(BuildOperationContext context) {
-                CompilationState previousCompileState = previousCompileStateCache.get();
-                IncrementalCompileSourceProcessor processor = incrementalCompileFilesFactory.files(previousCompileState);
+            public IncrementalCompilation call(@Nonnull BuildOperationContext context) {
+                IncrementalCompileSourceProcessor processor = incrementalCompileFilesFactory.files(previousCompileStateCache.get());
                 for (File sourceFile : sourceFiles) {
                     processor.processSource(sourceFile);
                 }
                 return processor.getResult();
             }
 
+            @Nonnull
             @Override
             public BuildOperationDescriptor.Builder description() {
-                ProcessSourceFilesDetails operationDetails = new ProcessSourceFilesDetails(sourceFiles.size());
-                return BuildOperationDescriptor
-                    .displayName("Processing source files")
-                    .details(operationDetails);
```

Comment:
this was never displayed as to string method does not display the field like when using a record or data class.